src/stories/4-elements/accordion/accordion.stories.tsx

```diff
@@ -0,0 +1,243 @@
+import { action } from "@storybook/addon-actions";
+import { ArgsTable, Description, Heading, PRIMARY_STORY, Stories, Title } from "@storybook/addon-docs";
+import { Meta, StoryFn } from "@storybook/react";
+import { useEffect, useRef } from "react";
+import { IFrontendEngineRef } from "../../../components";
+import { IAccordionSchema } from "../../../components/elements/accordion/types";
+import { CommonFieldStoryProps, DefaultStoryTemplate, FrontendEngine, SUBMIT_BUTTON_SCHEMA } from "../../common";
+
+const meta: Meta = {
+	title: "Element/Accordion",
+	parameters: {
+		docs: {
+			page: () => (
+				<>
+					<Title>Accordion</Title>
+					<Description>
+						Wrapping component that must be rendered as a direct descendant of `accordion` uiType.
+					</Description>
+					<Heading>Props</Heading>
+					<ArgsTable story={PRIMARY_STORY} />
+					<Stories includePrimary={true} title="Example" />
+				</>
+			),
+		},
+	},
+	argTypes: {
+		...CommonFieldStoryProps("accordion", true),
+		children: {
+			description: "Elements that are the descendant of this component",
+			table: {
+				type: { summary: "TFrontendEngineFieldSchema" },
+			},
+			type: { name: "object", value: {}, required: true },
+		},
+		button: {
+			description:
+				"<div>Define button-related settings.</div><ul><li><strong>label</strong>: Label of the button (Default: Edit)</li></ul>",
+			table: {
+				type: {
+					summary: "{ label: string }",
+				},
+				defaultValue: { summary: "{ label: 'Edit' }" },
+			},
+		},
+		collapsible: {
+			description: "Specifies if the contents can be collapsed or expanded",
+			table: {
+				type: {
+					summary: "boolean",
+				},
+				defaultValue: { summary: false },
+			},
+			control: {
+				type: "boolean",
+			},
+		},
+		expanded: {
+			description: "Specifies if the component is to be unfolded to reveal the items",
+			table: {
+				type: {
+					summary: "boolean",
+				},
+				defaultValue: { summary: true },
+			},
+			control: {
+				type: "boolean",
+			},
+		},
+		title: {
+			description: "A name of the purpose of the element",
+			table: {
+				type: {
+					summary: "string",
+				},
+			},
+		},
+		displayState: {
+			description: "Specifies the display state which renders an icon",
+			table: {
+				type: {
+					summary: `"default" | "error" | "warning" | undefined"`,
+				},
+			},
+			options: ["default", "error", "warning", "undefined"],
+			control: {
+				type: "select",
+			},
+		},
+	},
+};
+export default meta;
+
+const EventTemplate = <T, U = string>(id: string, eventName: string) =>
+	((args) => {
+		const formRef = useRef<IFrontendEngineRef>();
```

Comment:
use `// eslint-disable-next-line react-hooks/rules-of-hooks` to hide the eslint error
do the same for `useEffect`

Reply:
this is not done?

I am referring to the [react-hooks/rules-of-hooks](https://reactjs.org/docs/hooks-rules.html) eslint error with `useRef` and `useEffect`, they should be suppressed

Reply:
sorry missed out the useRef